packages/contracts-bedrock/test/DataAvailabilityChallenge.t.sol

```diff
@@ -0,0 +1,496 @@
+        vm.expectRevert(abi.encodeWithSelector(DataAvailabilityChallenge.ChallengeWindowNotOpen.selector));
+        dac.challenge(challengedBlockNumber, challengedHash);
+    }
+
+    function testChallengeFailAfterChallengeWindow() public {
+        uint256 challengedBlockNumber = 1;
+        bytes32 challengedHash = "some hash";
+
+        // Move to block after the challenge window
+        vm.roll(challengedBlockNumber + dac.challengeWindow() + 1);
+
+        // Challenge fails because the block number is after the challenge window
+        dac.deposit{ value: dac.bondSize() }();
+        vm.expectRevert(abi.encodeWithSelector(DataAvailabilityChallenge.ChallengeWindowNotOpen.selector));
+        dac.challenge(challengedBlockNumber, challengedHash);
+    }
+
+    function testResolveSuccess(
+        address challenger,
+        address resolver,
+        bytes memory preImage,
+        uint256 challengedBlockNumber,
+        uint256 resolverRefundPercentage
+    )
+        public
+    {
+        // Assume neither the challenger nor resolver is address(0)
+        vm.assume(challenger != address(0));
+        vm.assume(resolver != address(0));
+
+        // Assume the resolver refund percentage is valid
+        vm.assume(resolverRefundPercentage <= 100);
+
+        // Set the gas price to a non-zero value to test bond distribution logic
+        vm.txGasPrice(2);
+
+        // Change the resolver refund percentage
+        vm.prank(DAC_OWNER);
+        dac.setResolverRefundPercentage(resolverRefundPercentage);
+
+        // Assume the block number is not close to the max uint256 value
+        vm.assume(challengedBlockNumber < type(uint256).max - dac.challengeWindow() - dac.resolveWindow());
+        bytes32 challengedHash = keccak256(preImage);
+
+        // Move to block after challenged block
+        vm.roll(challengedBlockNumber + 1);
+
+        // Challenge the hash
+        uint256 bondSize = dac.bondSize();
+        vm.deal(challenger, bondSize);
+        vm.prank(challenger);
+        dac.challenge{ value: bondSize }(challengedBlockNumber, challengedHash);
+
+        // Resolve the challenge
+        vm.prank(resolver);
+        dac.resolve(challengedBlockNumber, challengedHash, preImage);
+
+        // Expect the challenge to be resolved
+        (address _challenger, uint256 _lockedBond, uint256 _startBlock, uint256 _resolvedBlock) =
+            dac.challenges(challengedBlockNumber, challengedHash);
+
+        assertEq(_challenger, challenger);
+        assertEq(_lockedBond, 0);
+        assertEq(_startBlock, block.number);
+        assertEq(_resolvedBlock, block.number);
+        assertEq(uint8(dac.getChallengeStatus(challengedBlockNumber, challengedHash)), uint8(ChallengeStatus.Resolved));
+
+        // Assert challenger balance after bond distribution
+        uint256 resolutionCost =
+            (dac.fixedResolutionCost() + preImage.length * dac.variableResolutionCost()) * tx.gasprice;
```

Comment:
In a world without 1559, a block builder could include a tx with a 0 gas price and break this. I believe its not possible to do this after 1559, even with legacy txs. This is nice because it includes the tip, while `block.basefee` doesn't

Reply:
Were you particular about using `tx.gasprice` over `block.basefee`? I worry that the semantics of `tx.gasprice` are different per tx type so new tx types in the future might allow for a loop hole here. `block.basefee` seems less able to game but doesn't capture the costs of when the blocks are saturated and txs compete on tips

Reply:
> a block builder could include a tx with a 0 gas price and break this

How would it break? in that case the effective fee the resolver pays would actually be 0 no?

The idea here is to estimate the total amount of gas the resolver had to pay to resolve the challenge, if we used `block.basefee` we'd ignore the priority fee the resolver also had to pay

Reply:
Before eip1559, you could include a tx that had 0 gas price. With 1559, you cannot because of the basefee requirement. You multiple by `tx.gasPrice` at the end so it would be `x * 0 = 0`

Reply:
I agree that `tx.gasPrice` takes into account the priority fee (tip). For the fault proof implementation, we have been using `block.basefee` over `tx.gasPrice`

cc @Inphi

Reply:
ok what do you think about `require(tx.gasPrice >= block.basefee);` ?

Reply:
Even if `tx.gasprice` is 0, that would mean the resolver paid no fee to resolve the transaction, so the whole bond would be refunded to the challenger. That seems like the intended behavior, could you elaborate why we would want to prevent this?

Reply:
I'm not sure including the tip is great here as it can be subject to manipulation. We should prevent the cost from being artificially inflated due to a high tip. 
The base fee already captures market conditions at the time of resolution. Anything else is just extra.

Reply:
changed to `block.basefee`